packages/client/lib/sync/fullsync.ts

```diff
@@ -93,15 +93,15 @@ export class FullSynchronizer extends Synchronizer {
    */
   async syncWithPeer(peer?: Peer): Promise<boolean> {
     // eslint-disable-next-line no-async-promise-executor
-    return await new Promise(async () => {
-      if (!peer) return false
+    return await new Promise(async (resolve, reject) => {
+      if (!peer) return resolve(false)
```

Comment:
Still no `Promise` expert, what would happen if we skip this whole `Promise` wrapping? 🤔

Generally this whole Promise resolving has gotten less important with the extraction of the sync status determination by listening and acting to chain updates (I already thougth about just returning `Promise<void>` on this function).

We just need to make sure that in the [Sync.start()](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/client/lib/sync/sync.ts#L94) call no concurrent `sync()` calls are triggered.

Reply:
Update: tested this with removing the Promise wrapper, this then stops at another test in `sync.spec.ts` unit test, couldn't investigate further but still not completely getting why (if) this is needed.

Reply:
I believe it's part of some promise chain that awaits for results, but I'm sure we could find an alternative architecture, I think it is probably a bit redundant now since it just awaits the SYNC'd event we can probably just listen for that wherever this promise is functionally being used.